packages/uicore/src/components/ExpandingSearchInput/ExpandingSearchInput.tsx

```diff
@@ -114,7 +114,7 @@ export function ExpandingSearchInput(
       const text = event.target.value
       setValue(text)
       timerRef.current = setTimeout(() => {
-        propsOnChange?.(text)
+        propsOnChange?.(text?.trim())
```

Comment:
What happens when user wants to search for
'Hello world'
If user takes a pause after typing 'hello ', will the space remain in the input or removed?

Reply:
inputs will not be modified (those are managed by setValue)
and for the onChange - value - (used for handling search)

let say space = _
hello_  or hello --> will search everything with hello or hello_ 

but as soon as the user types another char lets say `w` -> hello_w then it will only give `hello world` as result.

additionally change in `propsOnChange` here on uicore is similar to adding .trim at ngui

Reply:
I understand
So the requirement isto trim it everywhere?

Reply:
I see value is controlled in ExpandingSearchInput, rather than the parent, so this should wok fine.
We take defaultValue from parent